From: Tomasz Buchert <tomasz@debian.org>
Date: Sun, 5 Nov 2023 13:31:57 +0100
Subject: build static libraries by default

---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 61378cd..8dec247 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,7 +102,7 @@ unset(LOG2_RES)
 set(BROTLI_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/c/include")
 mark_as_advanced(BROTLI_INCLUDE_DIRS)
 
-set(BROTLI_LIBRARIES_CORE brotlienc brotlidec brotlicommon)
+set(BROTLI_LIBRARIES_CORE brotlienc brotlidec brotlicommon brotlienc-static brotlidec-static brotlicommon-static)
 set(BROTLI_LIBRARIES ${BROTLI_LIBRARIES_CORE} ${LIBM_LIBRARY})
 mark_as_advanced(BROTLI_LIBRARIES)
 
@@ -122,12 +122,15 @@ endif()
 
 file(GLOB_RECURSE BROTLI_COMMON_SOURCES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} c/common/*.c)
 add_library(brotlicommon ${BROTLI_COMMON_SOURCES})
+add_library(brotlicommon-static STATIC ${BROTLI_COMMON_SOURCES})
 
 file(GLOB_RECURSE BROTLI_DEC_SOURCES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} c/dec/*.c)
 add_library(brotlidec ${BROTLI_DEC_SOURCES})
+add_library(brotlidec-static STATIC ${BROTLI_DEC_SOURCES})
 
 file(GLOB_RECURSE BROTLI_ENC_SOURCES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} c/enc/*.c)
 add_library(brotlienc ${BROTLI_ENC_SOURCES})
+add_library(brotlienc-static  STATIC ${BROTLI_ENC_SOURCES})
 
 # Older CMake versions does not understand INCLUDE_DIRECTORIES property.
 include_directories(${BROTLI_INCLUDE_DIRS})
